fhir-persistence-jdbc/src/main/java/org/linuxforhealth/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -1002,6 +1003,56 @@ public MultiResourceResult search(FHIRPersistenceContext context, Class<? extend
+     * the expected first and last search results of the adjacent pages.
+     * Validate the input expected first resource Id and last resource Id with the search results.
+     * If the expected first resource Id or last resource Id do not match with the search results then add a OperationOutcome 
+     * with Issue Severity Warning to the result Builder.
+     * @param resourceDTOList the list of 'match' resources
+     * @param searchContext the current search context
+     * @param resultBuilder The MultiResourceResult builder
+     * @return the list of filtered resources after removing the additional resources.
+     */
+    private List<org.linuxforhealth.fhir.persistence.jdbc.dto.Resource>
+        validateExpectedSearchPagingResults(List<org.linuxforhealth.fhir.persistence.jdbc.dto.Resource> resourceDTOList, FHIRSearchContext searchContext, Builder resultBuilder) {
+        org.linuxforhealth.fhir.persistence.jdbc.dto.Resource firstResourceResult = null;
+        org.linuxforhealth.fhir.persistence.jdbc.dto.Resource lastResourceResult = null;
+        if (resourceDTOList != null) {
+            if (resourceDTOList.size() > 0 && searchContext.getPageNumber() != 1) {
+                resultBuilder.expectedPreviousId(resourceDTOList.get(0).getLogicalId());
+                resourceDTOList.remove(0);
+            } 
+            if (resourceDTOList.size() > searchContext.getPageSize()) {
+                resultBuilder.expectedNextId(resourceDTOList.get(resourceDTOList.size() - 1).getLogicalId());
+                resourceDTOList.remove(resourceDTOList.size() - 1);
+            }
+            if (resourceDTOList.size() > 0) {
+                firstResourceResult = resourceDTOList.get(0);
+                lastResourceResult = resourceDTOList.get(resourceDTOList.size() - 1);
+            }
+            if (firstResourceResult != null && searchContext.getFirstId() != null && !searchContext.getFirstId().equals(firstResourceResult.getLogicalId())) {
+                searchContext.addOutcomeIssue(OperationOutcome.Issue.builder()
+                    .severity(IssueSeverity.WARNING)
+                    .code(IssueType.CONFLICT)
+                    .details(CodeableConcept.builder()
+                        .text(string("Pages have shifted; check previous pages for changed results"))
+                        .build())
+                    .build());
+            }
+            if (lastResourceResult != null && searchContext.getLastId() != null && !searchContext.getLastId().equals(lastResourceResult.getLogicalId())) {
+                searchContext.addOutcomeIssue(OperationOutcome.Issue.builder()
+                    .severity(IssueSeverity.WARNING)
+                    .code(IssueType.CONFLICT)
+                    .details(CodeableConcept.builder()
+                        .text(string("Pages have shifted; check next pages for changed results"))
```

Comment:
The change in the sort order could come from anywhere - the next page, the current page or any previous page. I think in both cases we should simply say "check pages for changed results".

Reply:
This has been fixed as suggested.